doc/build-versioned-docs.sh

```diff
@@ -73,6 +73,8 @@ END
     # Actually building the docs for a given language and version
     sphinx-build -b html source/ build/html/${current_version}/${current_language} -A lang=True -D language=${current_language}
 
+    git clean -fd
```

Comment:
Should we include a comment here about what this does? If I understand correctly, the purpose is to remove any unwanted files that's not part of the checked-out repository tag. Since we start from the oldest branch and progressively check out newer tags, any folder/files structure that differ from newer branches will cause a `The following untracked working tree files would be overwritten by checkout:` error. I'm still fuzzy on the details.

Reply:
@chongshenng I added a small comment, wdyt? Should I be more verbose?

Reply:
@charlesbvll Yes, that works!